### PR TITLE
ghc 8.10.1

### PR DESCRIPTION
**Pull request by fishtreesugar**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by SMillerDev** [comment]
Does anything in homebrew depend on GHC@8.8?

**Comment by Bo98** [comment]
I wouldn't be surprised if something did before this pull request is merged. See how many use ghc@8.6.

File: Formula/ghc@8.8.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+require "language/haskell"
+
+class Ghc < Formula
```
Comment (Bo98) [inline Formula/ghc@8.8.rb]:
```suggestion
class GhcAT88 < Formula
```

File: Formula/ghc@8.8.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  desc "Glorious Glasgow Haskell Compilation System"
+  homepage "https://haskell.org/ghc/"
+  url "https://downloads.haskell.org/~ghc/8.8.3/ghc-8.8.3-src.tar.xz"
+  sha256 "e0dcc0aaf3e234c5978f29e6df62947e97720ab404ec0158343df211c5480f89"
+
+  bottle do
+    sha256 "efba70e7b2639df91b366fb54580b2de9cf8b774c367e4eeadf662bf8b6c7696" => :catalina
+    sha256 "6ce78524676be247d1729f219faea5fc0c91f084f48fce18d77d590d7f9d1a1b" => :mojave
+    sha256 "8724b8bd4409b1500ef78616a3edce4843c8c47e540605224458f6ff82dc213d" => :high_sierra
+  end
```
Comment (Bo98) [inline Formula/ghc@8.8.rb]:
```suggestion
```

ghc@8.8 counts as a new formula so omit the bottle block here.

File: Formula/ghc@8.8.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+
+    resource "cabal" do
+      url "https://hackage.haskell.org/package/cabal-install-3.0.0.0/cabal-install-3.0.0.0.tar.gz"
+      sha256 "a432a7853afe96c0fd80f434bd80274601331d8c46b628cd19a0d8e96212aaf1"
+    end
+  end
```
Comment (Bo98) [inline Formula/ghc@8.8.rb]:
```suggestion
  keg_only :versioned_formula
```

Versioned formula should not have a `head` block. They also need to be keg only.

**Review by Bo98 — COMMENTED** [review]
Question: what's the recommended version for the bootstrap (`resource "binary"`)? Is it still 8.6.5?

**Comment by fishtreesugar** [comment]
> Question: what's the recommended version for the bootstrap (resource "binary")? Is it still 8.6.5?

cc @bgamari

**Comment by fishtreesugar** [comment]
According to [release note of 8.8.3](https://www.haskell.org/ghc/blog/20200224-ghc-8.8.3-released.html), at least 8.8.3 could bootstrap itself.

**Comment by Bo98** [comment]
* `agda` - requires ghc@8.8 as is. See if it builds without `cpphs` though since it appears to be optional now.
* `allureofthestars` - might work when we retry the CI. `paralell` on Hackage has since been updated.
* `bench` - might work when we retry the CI. `paralell` on Hackage has since been updated.
* `cedille` - same situation as `agda`.
* `cgrep` requires ghc@8.8.
* `cmt` - possibly requires ghc@8.8? Not entirely clear.
* `dhall-lsp-server` - requires [this patch](https://github.com/dhall-lang/dhall-haskell/commit/8761fab60700b139b5fe9524732fb0602356aa6a.patch?full_index=1). Otherwise, `ghc@8.8` can be used if the patch doesn't apply.
* `futhark` - might work when we retry the CI. `zlib` on Hackage has since been updated.
* `highlighting-kate` - not entirely clear what GHC 8.10 has to do with this breakage.
* `hledger` - requires ghc@8.8.
* `hlint` - requires ghc@8.8.
* `komposition` - requires ghc@8.8.
* `mighttpd2` - requires ghc@8.8.
* `pandoc-citeproc` - requires ghc@8.8.
* `shellcheck` - requires ghc@8.8.
* `shelltestrunner` - requires ghc@8.8.
* `taskell` - requires ghc@8.8.

**Comment by bayandin** [comment]
I've stopped a build on CI since @Bo98's comment hasn't been addressed yet

**Comment by athas** [comment]
`futhark.rb` will require GHC 8.8 because it explicitly pins s specific Hackage index-state, so it will not pick up the newer `zlib`.  That's the downside of reproducibility.

**Comment by Bo98** [comment]
We prefer pinned versions anyway. So that's fine.

**Comment by athas** [comment]
The newest futhark.rb should work with both 8.8 and 8.10, so that should be one less special case.

**Comment by fishtreesugar** [comment]
I checked the build log and I don't know why CI jobs failed.

**Comment by fishtreesugar** [comment]
Tests failed with mysterious messages, like:

- agda-stdlib failed with wrong base constrains:
```
2020-04-03T01:49:59.0752460Z Resolving dependencies...
2020-04-03T01:49:59.0752700Z cabal: Could not resolve dependencies:
2020-04-03T01:49:59.0753660Z [__0] trying: lib-1.3 (user goal)
2020-04-03T01:49:59.0753900Z [__1] next goal: base (dependency of lib)
2020-04-03T01:49:59.0755190Z [__1] rejecting: base-4.14.0.0/installed-4.1... (conflict: lib =>
2020-04-03T01:49:59.0755530Z base>=4.9.0.0 && <4.14)
2020-04-03T01:49:59.0756520Z [__1] rejecting: base-4.12.0.0, base-4.11.1.0, base-4.11.0.0, base-4.10.1.0,
2020-04-03T01:49:59.0758850Z base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
2020-04-03T01:49:59.0759910Z base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
2020-04-03T01:49:59.0761350Z base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
2020-04-03T01:49:59.0762520Z base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
2020-04-03T01:49:59.0763260Z base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
2020-04-03T01:49:59.0765220Z non-upgradeable package requires installed instance)
2020-04-03T01:49:59.0765510Z [__1] fail (backjumping, conflict set: base, lib)
2020-04-03T01:49:59.0765730Z After searching the rest of the dependency tree exhaustively, these were the
2020-04-03T01:49:59.0766800Z goals I've had most trouble fulfilling: base, lib
2020-04-03T01:49:59.0767110Z Note: when using a sandbox, all packages are required to have consistent
2020-04-03T01:49:59.0767770Z dependencies. Try reinstalling/unregistering the offending packages or
2020-04-03T01:49:59.0769000Z recreating the sandbox.
```
and some formula failed with
```
openFile: does not exist (No such file or directory)
```

**Comment by Bo98** [comment]
Wrong base usually means that formula is not compatibile with the GHC you are using.

The "openFile: does not exist" happens if there is any network hitch while Cabal is downloading dependencies. It's annoying, but there's not much that can be done so you can ignore that, unless it happens with one of the formulae you are modifying in this pull request.

**Pull request closed without merging**